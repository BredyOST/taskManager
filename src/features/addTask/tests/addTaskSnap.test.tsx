import {fireEvent, render} from "@testing-library/react";
import AddTask from "../addTask.tsx";

jest.mock('../../../app/axios/taskApi/taskApi.ts', () => ({
    addTask: jest.fn()
}));

jest.mock('../../../shared/hooks/useTasks.ts', () => ({
    useTasks: () => ({
        refetch: jest.fn(),
    })
}))
jest.mock('../../../app/assets/svg/arrowDown.svg', () => 'svg');

describe('addTasks', () => {
    test('match snapShoot', () => {
        const {asFragment} = render(<AddTask openList={true} showListsTasks={() => {}}/>)
        expect(asFragment()).toMatchSnapshot();
    })

    test('handles task input and submit', () => {
        const {getByPlaceholderText} = render(<AddTask openList={true} showListsTasks={() => {}}/>)

        const input = getByPlaceholderText('ENTER TASK') as HTMLInputElement;

        // Имитация ввода текста в поле
        fireEvent.change(input, { target: { value: "new task" }})
        expect(input.value).toBe('new task');

        fireEvent.keyDown(input, { key: 'Enter' , code: 'Enter'})

        const {addTask} = require('../../../app/axios/taskApi/taskApi.ts')
        expect(addTask).toHaveBeenCalledTimes(1)
        expect(addTask).toHaveBeenCalledWith({description: 'new task', completed: false})

    })

})