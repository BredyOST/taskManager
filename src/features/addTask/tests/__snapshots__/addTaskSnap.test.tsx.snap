// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addTasks match snapShoot 1`] = `
<DocumentFragment>
  <div
    class="sc-egkSDF fBEUlG"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary sc-dntaoT bKQbZP css-1emzoey-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      id=":r0:"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiLoadingButton-label css-27yv9x-MuiLoadingButton-label"
      >
        <svg
          class="sc-fAUdSK jazaIM"
        />
      </span>
    </button>
    <div
      class="sc-gtLWhw gwxGCf"
    >
      <div
        class="MuiFormControl-root MuiTextField-root sc-blHHSb iQEfZM css-1xp5r68-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-5h82ro-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input css-1yrc8ca-MuiInputBase-input-MuiInput-input"
            id=":r1:"
            placeholder="ENTER TASK"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary sc-kLhKbu iQyTpg css-1emzoey-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      id=":r2:"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiLoadingButton-label css-27yv9x-MuiLoadingButton-label"
      >
        add
      </span>
    </button>
  </div>
</DocumentFragment>
`;
